File: java/socketd/src/main/java/org/noear/socketd/transport/core/ChannelInternal.java
package org.noear.socketd.transport.core;

import org.noear.socketd.transport.stream.StreamInternal;

import java.util.function.BiConsumer;

/**
 * 通道内部接口
 *
 * @author noear
 * @since 2.0
 */
public interface ChannelInternal extends Channel {
    /**
     * 设置会话
     */
    void setSession(Session session);

    /**
     * 更新最后活动时间
     */
    void setLiveTimeAsNow();

    /**
     * 更新告警代码
     * */
    void setAlarmCode(int alarmCode);

    /**
     * 获取流
     */
    StreamInternal getStream(String sid);

    /**
     * 当打开时
     */
    void onOpenFuture(BiConsumer<Boolean, Throwable> future);

    /**
     * 执行打开时
     */
    void doOpenFuture(boolean isOk, Throwable error);
}

File: python/socketd/socketd/transport/core/ChannelInternal.py
from abc import ABC, abstractmethod
from typing import Callable, Optional

from socketd.transport.core.Channel import Channel
from socketd.transport.core.Costants import Function
from socketd.transport.core.Session import Session


class ChannelInternal(Channel, ABC):
    @abstractmethod
    def set_session(self, session: Session): ...

    @abstractmethod
    def set_live_time_as_now(self):...

    @abstractmethod
    def set_alarm_code(self, alarm_code:int):...

    @abstractmethod
    def get_stream(self, sid: str): ...

    @abstractmethod
    def on_open_future(self, future: Function): ...

    @abstractmethod
    def do_open_future(self, is_ok: bool, e: Optional[Exception]): ...

